R/derive_vars_query.R

```diff
@@ -62,25 +62,17 @@
 #' )
 #' derive_vars_query(adae, queries)
 derive_vars_query <- function(dataset, dataset_queries) { # nolint: cyclocomp_linter
-  assert_data_frame(dataset_queries)
-  assert_valid_queries(dataset_queries, queries_name = deparse(substitute(dataset_queries)))
+  source_vars <- unique(dataset_queries$SRCVAR)
   assert_data_frame(dataset,
-    required_vars = exprs(!!!syms(unique(dataset_queries$SRCVAR))),
+    required_vars = chr2vars(source_vars),
     optional = FALSE
   )
 
   # check optionality of TERMNUM or TERMCHAR based on SRCVAR type
-  srcvar_types <- unique(vapply(dataset[dataset_queries$SRCVAR], typeof, character(1)))
-  if (length(srcvar_types) == 1 && srcvar_types == "character") {
-    assert_data_frame(dataset_queries, required_vars = exprs(TERMCHAR))
-  }
-  if (length(srcvar_types) == 1 && srcvar_types %in% c("integer", "double")) {
-    assert_data_frame(dataset_queries, required_vars = exprs(TERMNUM))
-  }
-  if (length(srcvar_types) > 1 &&
-    "character" %in% srcvar_types &&
-    ("integer" %in% srcvar_types || "double" %in% srcvar_types)) {
-    assert_data_frame(dataset_queries, required_vars = exprs(TERMCHAR, TERMNUM))
+  termvars <- exprs(character = TERMCHAR, integer = TERMNUM, double = TERMNUM)
+  expected_termvars <- unique(termvars[srcvar_types])
+  assert_data_frame(dataset_queries, required_vars = c(exprs(PREFIX, GRPNAME, SRCVAR), expected_termvars))
+  if (length(srcvar_types) > 1) {
```

Comment:
It is safer to use `expected_termvars` instead of `srcvar_type` because we could have `srcvar_type = c("integer", "double")`.

Reply:
it has been swapped